pkgs/development/tools/spirv-tools/default.nix

```
@@ -7,22 +7,22 @@ spirv_sources = {
   tools = fetchFromGitHub {
     owner = "KhronosGroup";
     repo = "SPIRV-Tools";
-    rev = "1a283f41ed09e31cd720744f904af3d823ceddbf";
-    sha256 = "1z65wglg081pri9rmiyydvppgd67qr269ppphy4yhg2wg81gg72c";
+    rev = "9bfe0eb25e3dfdf4f3fd86ab6c0cda009c9bd661";
+    sha256 = "1spfii4zib1lmz40vnz1d1cr6is7q2n7rvar1vfzi33gwrn8rqhi";
   };
   headers = fetchFromGitHub {
     owner = "KhronosGroup";
     repo = "SPIRV-Headers";
-    rev = "ff684ffc6a35d2a58f0f63108877d0064ea33feb";
-    sha256 = "0ypjx61ksr6vda2iy3kxhyjia5qxf0x4qa4jij0giw9x5rsnga6g";
+    rev = "d5b2e1255f706ce1f88812217e9a554f299848af";
+    sha256 = "18530mpa030ckjhn78z9vbfd35l5jgh3mg22ra6k8gw8zx4wjhsl";
   };
 };
 
 in
 
 stdenv.mkDerivation rec {
   name = "spirv-tools-${version}";
```

```suggestion
  name = "spirv-tools-${version}-unstable";
```

(same goes for glslang)

> If a package is not a release but a commit from a repository, then the version part of the name must be the date of that (fetched) commit. The date must be in "`YYYY-MM-DD`" format. **Also append "`unstable`" to the name - e.g., "`pkgname-unstable-2014-09-23`".**

https://nixos.org/nixpkgs/manual/#sec-package-naming

---

This change is fine with me, but I'd like to point out that the release status of these packages is quite strange: we are using a released version of `vulkan-validation-layers`, and that package's source identifies a specific release of `glslang` to be good. Likewise, `glslang` identifies a specific revision of  `spirv-tools`.

This puts us in a troubling spot, because if we rely on releases of the upstream packages (e.g. `spirv-tools`) rather than revisions identified by the downstream packages (e.g. `vulkan-validation-layers`), things break (I experienced this myself in preparing this little patch). But if we use all the "known good" revisions starting with a downstream package (like `vulkan-validation-layers`) and follow the nixpkgs naming schemes, we will have a bunch of `unstable` versions of things despite these being the "known good" versions required by the downstream packages.

Can we at least get @Ralith to chime in, since they are the maintainer? I am fine with appending `unstable`, but have a very slight preference to not do so because it has an inaccurate connotation. On the other hand, a good consequence of appending `unstable` to these "known good" versions is that it would make room for the released versions if somebody needed them, though this does seem ripe for confusion.

---

Corollary: we may want to add the `unstable` suffix to a bunch of these packages that flow from the `known_good.json` files. Should we do that in this PR or split it out into a separate task?

---

Oh, I see. In that case, I don't think `unstable` is right here after all – I didn't realise the strange way the versioning worked.

---

I don't have a strong opinion here; vulkan tools/validation layers selecting a specific glslang certainly seems like a declaration that it's stable enough for their purposes, but e.g. shaderc uses entirely different commits, and there doesn't seem to be any pretense of a conventional stability guarantee.